src/GraphQL/NameAttribute.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using GraphQL.Conversion;
+using GraphQL.Types;
+
+namespace GraphQL
+{
+    /// <summary>
+    /// Specifies a GraphQL type name for a CLR class, or a field name for a property.
+    /// Note that the specified name will be translated by the schema's <see cref="INameConverter"/>.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Property, AllowMultiple = false, Inherited = false)]
+    public class NameAttribute : GraphQLAttribute
```

Comment:
A bit confused by `NameAttribute` vs `InputNameAttribute` vs `OutputNameAttribute`.
First modifies both type and field. Second and third only type.

Reply:
But `InputTypeAttribute` and `OutputTypeAttribute` work for type AND field 😕 .

Reply:
We can certainly make InputName and OutputName work for field also - easy. 👍 I didn’t because I couldn’t think of a reason for doing so. Whereas to use a Person clr type as an input and output type requires different names. Similarly, InputType and OutputType must both exist for a property if a single clr class is covering both input and output GraphQL classes.

Reply:
Done (need to add tests though)

Reply:
Added tests.